contexts/network-settings.tsx

```diff
@@ -64,120 +62,199 @@ export const NetworkSettingsProvider = ({ children }) => {
+    settings.parameters.disputableTime.validated = parametersValidations[2];
+    settings.parameters.percentageNeededForDispute.validated = parametersValidations[3];
+    settings.parameters.validated = parametersValidations.every(condition => condition);
+    //Theme
+    const colors = settings.theme?.colors;
+    
+    if (colors?.primary){
+      const similar = [];
+
+      similar.push(...isColorsSimilar({ label: "text", code: colors.text }, [
+      { label: "primary", code: colors.primary },
+      { label: "background", code: colors.background },
+      { label: "shadow", code: colors.shadow },
+      ]));
+
+      similar.push(...isColorsSimilar({ label: "background", code: colors.background }, [
+        { label: "success", code: colors.success },
+        { label: "danger", code: colors.danger },
+        { label: "warning", code: colors.warning },
+      ]));
+
+      if (!isSameSet(new Set(similar), new Set(settings.theme?.similar)))
+        settings.theme.similar = similar;
+    }
+
+    return settings;
+  }
+
+  function handerValidateForm(newState) {
+
+    const tokensLockedValidate = [
+      Fields.amount.validator(newState.tokensLocked?.locked, newState.tokensLocked?.needed)
+    ].every(condition => condition);
+      
+    const detailsValidate = [
+      newState.details.name.validated,
+      newState.details.fullLogo.validated,
+      newState.details.iconLogo.validated,
+    ].every(condition => condition);
+
+    const githubValidate = [
+        Fields.repository.validator(newState.github?.repositories),
+        newState.github?.botPermission,
+    ].every(condition => condition);
+    
+    newState.settings = handlerValidateSettings(newState.settings)
+
+    const settingsValidated = [
+    !newState.settings?.theme?.similar?.length,
+    newState.settings?.treasury?.validated,
+    newState.settings?.parameters?.draftTime?.validated,
+    newState.settings?.parameters?.disputableTime?.validated,
+    newState.settings?.parameters?.percentageNeededForDispute?.validated,
+    newState.settings?.parameters?.councilAmount?.validated,
+    ].every(condition => condition);
+    
+    const tokensValidated = [
+      newState.tokens?.settler?.trim() !== "",
+    ].every(condition => condition);
+    
+    newState.tokensLocked.validated = tokensLockedValidate;
+    newState.details.validated = detailsValidate;
+    newState.github.validated = githubValidate;
+    newState.settings.validated = !!settingsValidated;
+    newState.tokens.validated = tokensValidated;
+    newState.isSettingsValidated = [tokensLockedValidate,
+                                    detailsValidate,
+                                    githubValidate,
+                                    settingsValidated,
+                                    tokensValidated,
+    ].every(condtion=> condtion)
+
+    if(detailsValidate)
+      localStorage.setItem(`${localStorageKey}:${wallet.address}`, 
```

Comment:
Are we clearing this information when the user disconnect ? @clarkjoao

Reply:
No, only past 2 days or after create network.

Reply:
As we are clearing other data from localStorage on disconnect, i think we can do it for network settings data too

Reply:
I did a double check, and when we disconnect we are cleaning every items in localstorage. 🙂

<img width="511" alt="Screenshot 2022-09-28 at 21 01 28" src="https://user-images.githubusercontent.com/46800211/192877541-47c5a105-0392-42ff-b433-c9b2fb1fe85a.png">